crates/kira/tests/tick_accuracy.rs

```diff
@@ -0,0 +1,100 @@
+use kira::{
+	clock::{clock_info::WhenToStart, ClockHandle, ClockSpeed, ClockTime},
+	dsp::Frame,
+	manager::{
+		backend::mock::{MockBackend, MockBackendSettings},
+		AudioManager, AudioManagerSettings,
+	},
+	track::{
+		effect::{Effect, EffectBuilder},
+		TrackBuilder,
+	},
+};
+
+const SAMPLE_RATE: u32 = 48_000;
+const TICKS_PER_SECOND: f64 = 1000.0;
+const EVENT_TIME_TICKS: u64 = 500;
+
+struct TestEffect {
+	clock: ClockHandle,
+	ticks: u64,
+	frames: u32,
+}
+
+impl Effect for TestEffect {
+	fn process(
+		&mut self,
+		input: Frame,
+		_dt: f64,
+		clock_info_provider: &kira::clock::clock_info::ClockInfoProvider,
+		_modulator_value_provider: &kira::modulator::value_provider::ModulatorValueProvider,
+	) -> Frame {
+		self.frames += 1;
+		if let WhenToStart::Now = clock_info_provider.when_to_start(ClockTime {
+			clock: self.clock.id(),
+			ticks: self.ticks,
+		}) {
+			assert_eq!(self.ticks % EVENT_TIME_TICKS, 0);
```

Comment:
What is this assert used for? From what I can tell, `self.ticks` will always be a multiple of `EVENT_TIME_TICKS` because it's only ever incremented by `EVENT_TIME_TICKS`.

Reply:
You are right, it's mostly unnecessary. It would only fail, if `when_to_start` is incorrect.
If you want I can remove it.

Reply:
Yes, please. Also, can you add some comments explaining how the test works? It took me a while to understand it just by reading it.

Reply:
Did you have time to check the changes and do the comments work for you?

Reply:
That looks good! I might tweak some things later, but I'll go ahead and merge this in. Thank you!